### PR TITLE
Add flag to force-enable JS serving

### PR DESCRIPTION
**Pull request by sgammon**
For unclear reasons, `ibazel` doesn't detect our use of the `ibazel_live_reload` tag, and therefore never starts the live reload server. Fixes and closes bazelbuild/bazel-watcher#358.

When our server loads up and runs, it never sees the environment variable. If we try to load the `lrserver`'s `livereload.js` at the default port, it fails -- because the server never starts listening in the first place.

This adds a simple flag to `ibazel`: `force_live_reload`. When this flag is passed, `ibazel` starts up the `lrserver` in `Initialize` instead of during target scanning.

Changes enclosed:
- [x] Add `force_live_reload` flag, init `lrserver` in `Initialize` when passed
- [x] Elide scan for `ibazel_live_reload` tag when `-force_live_reload` is active (the server would already be on)

### CODE REVIEW

**Comment by achew22** [comment]
@sgammon, thanks for the PR! I'm definitely interested in making js live reload work for everyone!

Before we get too far down this road, let me describe my concerns. I'm concerned that if we adopt this approach, ibazel will move toward a thing that I've long been afraid of, "another app with an incompatible set of command line arguments that I can never remember". The value in having params encoded in `BUILD` files is remarkable from that perspective. Additionally, I'm concerned that there is something wrong in the live reload detection logic and I would REALLY like to figure that out since it's probably driving other people crazy.

As I understand it, you've got a `tag` set correctly on a target in your build graph. Is there any way you can make a runnable example that's available on github that exhibits this problem where it has the tag but the tag isn't detected by `ibazel`?

**Comment by sgammon** [comment]
hey there @achew22 - i agree, and i would be happy to diagnose it :) for now, though, we have deadlines and not having live-reload nerfs our use of `ibazel` pretty hard. we see serious slowness with the `bazel info` phase, too (it takes 3-5 mins to start the server), so we're hoping to just get this out of the way for ourselves and fix that, and then use the productivity gains to work back upstream and get things fixed/merged.

in the meantime, i've found the source of my earlier confusion about the env var showing up. the env var shows up fine, it's just that the tag is never detected, and so the server never starts. by adding this flag, the server starts fine, env var is there, server loads, all is well with the world.

our app is based on a homegrown OSS framework ([elide](https://github.com/sgammon/elide)), and if it applies to us, it likely applies to any targets wired by macros defined there. so i suspect i can make a minimal repro, and drop that here soon.

**Comment by achew22** [comment]
> for now, though, we have deadlines and not having live-reload nerfs our use of ibazel pretty hard.

Good old deadline pressure. Sorry to hear that!

> we see serious slowness with the bazel info phase, too (it takes 3-5 mins to start the server), so we're hoping to just get this out of the way for ourselves and fix that, and then use the productivity gains to work back upstream and get things fixed/merged.

That comment about the `bazel info` phase is really concerning to me WRT the configuration of your repo. From my experience at Google with Blaze and a [+2bn loc](https://www.wired.com/2015/09/google-2-billion-lines-codeand-one-place/) repo I got `bazel info` results back in much less than 5s.

> in the meantime, i've found the source of my earlier confusion about the env var showing up. the env var shows up fine, it's just that the tag is never detected, and so the server never starts. by adding this flag, the server starts fine, env var is there, server loads, all is well with the world.

Good to hear that this is alleviating the pressure. 

> our app is based on a homegrown OSS framework (elide), and if it applies to us, it likely applies to any targets wired by macros defined there. so i suspect i can make a minimal repro, and drop that here soon.

As a datapoint, at my work we have a fairly elaborate macro based framework for our servers and do not suffer from this.

I don't want you don't feel like I'm goalpost moving, so let me explicitly state my position. I'm 90% confident that I'm not going to merge this since it feels like we're hacking around either a bug in ibazel or a misconfiguration of your repo. If your launch pressure is strong enough, a strategy for advancing might be to build your own ibazel binary and distribute it with these changes. I would like to keep the conversation going so we can figure out if it is an ibazel or configuration problem though. A minimal repro would be really great for figuring this out.

**Comment by sgammon** [comment]
@achew22 forgive me, perhaps that deadline pressure is leaking through in my commentary, because i see your position, understand, and agree 😄

when i first submitted, i felt like this PR was a good complement to `nolive_reload`, but now i realize that is meant for suppression of the live reload feature _without_ requiring changes in code. so that does make sense and explains why a flag the other way might be different.

i also understand and agree with your point about there likely being a bug here that needs fixing, which should get fixed, rather than papering that behind a flag. so let me rephrase! lol

we _are_ distributing this, with the fix, as a local binary and using it for our needs. that's fine for now, our team is relatively small. i would love to put together a repro that forces this out into the open.

> Good old deadline pressure. Sorry to hear that!

thank you, beauty is painful or at least that's what they tell me

> That comment about the `bazel info` phase is really concerning to me WRT the configuration of your repo. From my experience at Google with Blaze and a +2bn loc repo I got bazel info results back in much less than 5s.

wow, that is really concerning. okay. it sounds to me like we're doing something absurd somewhere along the way. would the profiling tools built into `ibazel` help diagnose this, or would the bazel target graph be the first thing you'd reach for?

> As a datapoint, at my work we have a fairly elaborate macro based framework for our servers and do not suffer from this.

that is comforting and thank you for sharing. i suspect our use of macros is rather naive, i wired together the pieces of this framework while learning bazel's internals, which admittedly still confuse me greatly. intuitively, i figure this is related to flags in a `.bazelrc` somewhere, or those macros. where would you begin diagnosing, if different from the previous answer?

thank you, by the way, for your help on this. i'll get this repro up as soon as i can.
